### PR TITLE
[MM-20514] Migrates components/password reset send link to typescript

### PR DESCRIPTION
**Pull request by sikloidz**
#### Summary
Migrates files under `componentes/password_reset_send_link` to typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13726

### CODE REVIEW

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @sikloidz!

**Review by jgilliam17 — APPROVED** [review]
Thank you @sikloidz 
Tested, looks good to merge.
- Verified password reset link functionality - as expected.

File: components/password_reset_send_link/password_reset_send_link.tsx [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
-    emailInput = React.createRef();
+    resetForm = React.createRef<HTMLFormElement>();
+    emailInput = React.createRef<HTMLInputElement>();
 
-    handleSendLink = async (e) => {
+    handleSendLink = async (e: React.FormEvent) => {
         e.preventDefault();
 
-        const email = this.emailInput.current.value.trim().toLowerCase();
+        const email = this.emailInput.current!.value.trim().toLowerCase();
```
Comment (calebroseland) [inline components/password_reset_send_link/password_reset_send_link.tsx:37]:
```suggestion
        const email = this.emailInput.current?.value.trim().toLowerCase();
```
Minor:  I think `?` would be a bit more appropriate

File: components/password_reset_send_link/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
```
Comment (calebroseland) [inline components/password_reset_send_link/index.ts:3]:
```suggestion

import {ComponentProps} from 'react';
```

File: components/password_reset_send_link/index.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import {connect} from 'react-redux';
+import {sendPasswordResetEmail} from 'mattermost-redux/actions/users';
+import {GenericAction, ActionFunc} from 'mattermost-redux/types/actions';
+import {ServerError} from 'mattermost-redux/types/errors';
+
+import PasswordResetSendLink from './password_reset_send_link';
+
+type Actions = {
+    sendPasswordResetEmail: (emal: string) => Promise<{data: any; error: ServerError}>;
+}
```
Comment (calebroseland) [inline components/password_reset_send_link/index.ts:14]:
```suggestion
type Actions = type Actions = ComponentProps<typeof PasswordResetSendLink>['actions'];
```
Minor: could be de-duped

**Review by calebroseland — APPROVED** [review]
LGTM! Just two minor suggestions, but also fine as-is

File: components/password_reset_send_link/index.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {bindActionCreators, Dispatch, ActionCreatorsMapObject} from 'redux';
+import {connect} from 'react-redux';
+import {sendPasswordResetEmail} from 'mattermost-redux/actions/users';
+import {GenericAction, ActionFunc} from 'mattermost-redux/types/actions';
+import {ServerError} from 'mattermost-redux/types/errors';
```
Comment (calebroseland) [inline components/password_reset_send_link/index.ts:8]:
```suggestion
```

File: components/password_reset_send_link/password_reset_send_link.tsx [hunk 5]
```diff
@@ -1,41 +1,46 @@
...
 import {t} from 'utils/i18n.jsx';
 
-export default class PasswordResetSendLink extends React.PureComponent {
-    static propTypes = {
-        actions: PropTypes.shape({
-            sendPasswordResetEmail: PropTypes.func.isRequired,
-        }).isRequired,
+interface Props {
+    actions: {
+        sendPasswordResetEmail: (emal: string) => Promise<{data: any; error: ServerError}>;
```
Comment (calebroseland) [inline components/password_reset_send_link/password_reset_send_link.tsx:17]:
```suggestion
        sendPasswordResetEmail: (email: string) => Promise<{data: any; error: ServerError}>;
```

**Comment by hanzei** [comment]
@sikloidz Just want to touch bases and check if you have any questions about the feedback above?

**Comment by devinbinnie** [comment]
@hanzei I think we can just merge as-is.

**Comment by hanzei** [comment]
@calebroseland What do you think?

**Pull request merged**